lib/ruby_lsp/listeners/hover.rb

```diff
@@ -13,6 +13,7 @@ class Hover
           Prism::ConstantReadNode,
           Prism::ConstantWriteNode,
           Prism::ConstantPathNode,
+          Prism::GlobalVariableReadNode,
```

Comment:
We should also handle `GlobalVariableWriteNode`.

(there's also `GlobalVariableAndWriteNode` and a few others, but they are very rarely used).

Reply:
I'm wondering, after reading your feedback, if I missed this point with the indexer part as well.

```ruby
# a new comment, that will not be displayed in the hover
$DEBUG = "foo"
```

Here, on hover, if I'm not mistaken, we want to display both comments: the one from the RBS declarations and the one from our own implementation. If yes, then I need to handle all these kinds of nodes within `RubyIndexer::DeclarationListener`, not only on RBS side. 

WDYT?

Reply:
Also, I don't see any issues with shipping this PR and following it up with the rest of the nodes, if you prefer that approach.

Reply:
Yes, we probably should. We index comments for constants defined in the user's own code, so I would expect globals to behave the same way.

Reply:
(wrote that before I saw Vini had already replied)

Reply:
I would prefer to hold-off this one, create a separate PR focusing solely on indexing all kinds of global variables, and then adjust this one, taking advantage of all the additions. This way, we can keep concerns clearly separated, WDYT?

Reply:
It's very possible that the condition is no longer necessary. In the case of hovering over global assignments, we would need something along these lines:

```ruby
if ...
elsif target.is_a?(Prism::GlobalVariableWriteNode) && !covers_position?(target.name_loc, position)
 target = nil
end
```